PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -371,6 +395,7 @@ def nanoAOD_customizeCommon(process):
                                      addParticleNet=nanoAOD_addDeepInfoAK8_switch.nanoAOD_addParticleNet_switch,
                                      jecPayload=nanoAOD_addDeepInfoAK8_switch.jecPayload)
     (run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1).toModify(process, lambda p : nanoAOD_addTauIds(p))
+    (~(run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1)).toModify(process, lambda p : nanoAOD_addBoostedTauIds(p))
```

Comment:
@swozniewski in the CMSSW master we put the latest-greatest developments i.e., we do not protect against previews CMSSW versions. Please remove the modifiers here and elsewhere in this file. The modifiers would be need when you make the PR in CMSSW_106X.

Reply:
@gouskos I'm not sure how much we care in master, but as outlined during the meeting, previous miniAOD versions do not provide the correct input for boosted taus. Assuming that 11_X is not used with those, shall I still remove the modifiers? For the backport to 10_6_X, I should keep them, right?

Reply:
@swagata87 After the discussion in the meeting I understood the situation, which for the sake of completeness and bookeeping let me briefly summarise it here. The first miniAOD version that provides the correct inputs for BoostedTau is the UL miniAOD sent for production Dec 2020. 
As it is now, you do not want to include boostedTaus in previous campaigns, right? If yes, then I think is safer to keep this part of the code as it is also for the master. 

For 106X yes - you should keep them and add "run2_nanoAOD_devel". [Effectively, you want to always recalculate the inputs if not in "run2_nanoAOD_devel"] So I think it should look like:
`(~(run2_nanoAOD_94XMiniAODv1 | run2_nanoAOD_94X2016 | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1) & run2_nanoAOD_devel).toModify(process, lambda p : nanoAOD_addBoostedTauIds(p)) `
[but needs to be tested]

Reply:
Right, we think it's only safe to include them from the latest campaign on. Ok, so I'll keep it. Is there sth left for this PR to master (in case I missed sth)?
It's actually not 're'-calculating the inputs. If boosted taus are written out, we also need to run this ID. My feeling is that boosted taus should be produced independently from run2_nanoAOD_devel (but maybe I'm wrong about the purpose of this modifier). Could you please confirm what dependency is envisaged? Then I'll add it to the backport and could also open the PR for it.

Reply:
> Right, we think it's only safe to include them from the latest campaign on. Ok, so I'll keep it. Is there sth left for this PR to master (in case I missed sth)?

It looks good to me - we just need to run the standard nanoAOD tests, but there is an issue we are trying to solve before testing the PR. I would suggest to move to the 106X PR in the meantime

> It's actually not 're'-calculating the inputs. If boosted taus are written out, we also need to run this ID. My feeling is that boosted taus should be produced independently from run2_nanoAOD_devel (but maybe I'm wrong about the purpose of this modifier). Could you please confirm what dependency is envisaged? Then I'll add it to the backport and could also open the PR for it.

Thanks for the clarification! So then what you have there should work.

Reply:
I got a bit delayed by an incompatibility in 10_6_X and I had to include a fix. I'll create the backport PR now.